runtime/lua/vim/lsp/util.lua

```diff
@@ -352,17 +352,22 @@ local function get_line_byte_from_position(bufnr, position, offset_encoding)
 end
 
 --- Process and return progress reports from lsp server
----@private
+---
```

Comment:
why was `@private` removed ?

Reply:
What I mean is that this function is really helpful for displaying lsp related information in statuslines, hence added some documentation and removed `@private`

I assumed that adding documentation would be sufficient to make this function public? Since it seemed to me that it didn't do anything too tied to the internals of lsp functioning?

I maybe very wrong though, so do let me know if I am missing something.

Reply:
making it public means it needs to be moved to `vim.lsp.x()` directly. `vim.lsp.util` is private. Anything not private on vim.lsp.util is a mistake.

I don't see any usages of `get_progress_messages` in the codebase, let alone tests. So it needs test coverage also.

Reply:
if the function is useful, we can move forward with making it public. but the steps I've mentioned are needed to do that

Reply:
Alright, will do!

Reply:
We should definitely have some way for users to integrate the LSP progress messages into their statusline. Not sure if `get_progress_messages` is the best way for that - but it's currently being used for it, even if private and in the `util` package.

So unless there's a better proposal for retrieval of the progress messages I'd vote for promoting it to `vim.lsp.`

Reply:
What would your thoughts be on adding a `data` table to the autocmd event similar to how `LspAttach` and `LspDetach`? Then the user would retrieve the messages from `event.data.messages` or similar.
The problem I see with that is then if the user wanted to extract messages without waiting for the event they would have to force-trigger it using `nvim_exec_autocmds` or similar which could be an issue.

Reply:
I thought some more about this and now think we should actually deprecate + remove `get_progress_messages` and replace it with a `vim.lsp.status()` or something like that.

We should probably also stop storing the progress messages in `client.messages` per token but instead have a small ringbuffer. I think otherwise memory usage grows unbounded if the user doesn't consume them.

If users want to customize the progress display they can still override the handler.